app/imageUtils.js

```diff
@@ -0,0 +1,208 @@
+    }
+    return URL
+}
+
+async function hasIOPermissions(path) {
+    if(!fs.existsSync(path)) throw new Error('Invalid Path Parameter: Path does not exist.')
+    try {
+        await fs.promises.access(path, fs.constants.R_OK | fs.constants.W_OK)
+        return true
+    } catch(err) {
+        fs.stat(path, (_, stats) => {
+            logger.info(`Invalid R/W Permissions on path:${path + '\n'} ${'0' + (stats.mode & parseInt('777', 8)).toString(8)}`)
+        })
+        return false
+    }
+}
+
+function createRandomString(length) {
+    if(length <= 0 || !parseInt(length)) return
+    return crypto.randomBytes(parseInt(length)).toString('hex')
+}
+
+async function __saveImage(image, imageName, folder, imageWidths) {
+    try {
+        if(!image) throw new Error('Invalid Parameters on __saveImage: No Image Passed')
+        const webpImages = await createImages(image, imageName, imageWidths, folder, WEBP)
+        const jpegImages = await createImages(image, imageName, imageWidths, folder, JPEG)
+        const allImagesSaved = [...webpImages, ...jpegImages].every(result => !!result)
+        return allImagesSaved
+    } catch(err) {
+        logger.info(`Save Image: ${err}`)
+    }
+}
+
+async function createImages(image, imageName, imageWidths, folder, format) {
+    const responses = []
+    for(let width of imageWidths) {
+        const dirPath = getImageDirectory(path.join(folder, width))
+        const hasPermissions = await hasIOPermissions(dirPath)
+        if(!hasPermissions) throw new Error(`Error: Invalid Permissions at ${dirPath}`)
+        responses.push(await __saveImageToFile(image, `${imageName}.${format}`, dirPath, parseInt(width), format))
+    }
+    return responses
+}
+async function __saveImageToFile(image, imageName, dirPath, width, format) {
+    const filePath = path.join(dirPath, imageName)
+    if(format === WEBP) return await sharp(image.data).resize({width}).webp(WEBP_OPTIONS).toFile(filePath)
+    return await sharp(image.data).resize({width}).jpeg(JPEG_OPTIONS).toFile(filePath)
+}
+async function __getImage(res, imageName, folder, webpFormat, width) {
+    try {
+        if(!imageName || !folder) throw new Error('Error: Invalid parameters: ', imageName, folder)
+        const imageFormat = webpFormat ? WEBP : JPEG
+        const mimeType = webpFormat ? WEBP_MIMETYPE : JPEG_MIMETYPE
+
+        const dirPath = getImageDirectory(path.join(folder, width))
+        const hasPerms = await hasIOPermissions(dirPath)
+        if(!hasPerms) throw new Error(`Invalid R/W Permissions at ${dirPath}`)
+        
+        const filePath = path.join(dirPath, `${imageName}.${imageFormat}`)
+        if(!fs.existsSync(filePath)) throw new Error(`Error: ${filePath} does not exist`)
+        const imageBuffer = await fs.promises.readFile(filePath)
+        res.set('Content-Type', mimeType)
+        return res.send(imageBuffer)
+    } catch(err) {
+        logger.info(`__getImage: ${err}`)
```

Comment:
won't it display stack trace, logging out the function name might be unecessary

Reply:
Not all the time, it makes debugging significantly easier just to add this to the catch logs